### PR TITLE
Packager status page & build validating against it.

### PR DESCRIPTION
**Pull request by JustinCarmony**
Creating a packager status page so React can validate a proper packager instance is running on 8081.

See #257 for details on this bug.

The biggest thing in this PR is I have it perform an exit 2 in the build script if the check fails. This will cause the build to fail, they can click on the error and see a nice message. Not sure if there is a way to throw a warning instead.

Also, I broke the bash script into several lines, in the Xcode editor it looks fine but in the source code it looks less than ideal. We might want to break that out into it's own bash script that is called. Let me know if you want to do that.

### CODE REVIEW

**Comment by amasad** [comment]
I like it. @frantic want to also take a quick look. There's been a few people having issues with the port, see #257 #304

File: packager/packager.js [hunk 1]
```diff
@@ -150,6 +150,20 @@ function getDevToolsLauncher(options) {
   };
 }
 
+// A status page so the React/project.pbxproj build script
+// can verify that packager is running on 8081 and not 
+// another program / service.
+function statusPageMiddleware(req, res, next){
```
Comment (amasad) [inline packager/packager.js]: nit: space between `)` and `{`

File: packager/packager.js [hunk 2]
```diff
@@ -150,6 +150,20 @@ function getDevToolsLauncher(options) {
...
+  if(req.url == '/packager-status')
```
Comment (amasad) [inline packager/packager.js]: nit: spaces between `if` and `(` and `{` on the right please

Comment (amasad) [inline packager/packager.js]: let's also just make it `/status`

**Comment by JustinCarmony** [comment]
@amasad Done. :)

File: packager/packager.js [hunk 3]
```diff
@@ -150,6 +150,20 @@ function getDevToolsLauncher(options) {
   };
 }
 
+// A status page so the React/project.pbxproj build script
+// can verify that packager is running on 8081 and not 
+// another program / service.
+function statusPageMiddleware (req, res, next) {
+  if (req.url == '/status')
+  {
```
Comment (vjeux) [inline packager/packager.js]: nit: put the { in the previous line :)

```
if (req.url === '/status') {

} else {

}
```

File: packager/packager.js [hunk 4]
```diff
@@ -150,6 +150,20 @@ function getDevToolsLauncher(options) {
   };
 }
 
+// A status page so the React/project.pbxproj build script
+// can verify that packager is running on 8081 and not 
+// another program / service.
+function statusPageMiddleware (req, res, next) {
+  if (req.url == '/status')
```
Comment (vjeux) [inline packager/packager.js]: nit: triple =

**Comment by JustinCarmony** [comment]
Done. Is there a linter I can run? :P

**Comment by JustinCarmony** [comment]
@vjeux @amasad Anthing that I can do for this PR? I've had a few co-workers get bitten by this the other day.

Thanks!

**Comment by vjeux** [comment]
Sorry, our tooling is not good enough right now, an internal diff has been made and accepted. Hopefully it should land tomorrow.

**Comment by JustinCarmony** [comment]
@vjeux should I close this PR then?

**Comment by vjeux** [comment]
Not yet, we'll close it when it lands :)

File: packager/packager.js [hunk 5]
```diff
@@ -150,6 +150,19 @@ function getDevToolsLauncher(options) {
...
 
+// A status page so the React/project.pbxproj build script
+// can verify that packager is running on 8081 and not 
+// another program / service.
+function statusPageMiddleware (req, res, next) {
+  if (req.url === '/status') {
+    res.end('packager-status:running');
+  }
+  else
+  {
```
Comment (sahrens) [inline packager/packager.js]: nit: the curly braces should be on the same line as the else:

  } else {

File: packager/packager.js [hunk 6]
```diff
@@ -150,6 +150,19 @@ function getDevToolsLauncher(options) {
   };
 }
 
+// A status page so the React/project.pbxproj build script
+// can verify that packager is running on 8081 and not 
+// another program / service.
+function statusPageMiddleware (req, res, next) {
```
Comment (sahrens) [inline packager/packager.js]: nit: no space between function name and opening paren:

  function statusPageMiddleware(req, res, next) {

**Comment by sahrens** [comment]
@JustinCarmony: Can you address those nits real quick before we land internally?

**Comment by JustinCarmony** [comment]
@sahrens done.

**Comment by sahrens** [comment]
Great, this is now landing internally.

**Comment by amasad** [comment]
This landed and published in v0.3.3

**Comment by sahrens** [comment]
Check out linter.js

**Pull request closed without merging**